src/mqueryfront/src/components/ActionShowMatchContext.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useRef, useEffect } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faLightbulb } from "@fortawesome/free-solid-svg-icons";
+import Draggable from "react-draggable";
+
+const useClickOutside = (ref, callback) => {
+    const handleClick = (event) => {
+        if (ref.current && !ref.current.contains(event.target)) {
+            // lose focus (higher z-index) only if other modal was clicked
+            const modals = document.querySelectorAll(".modal");
+            const wasClicked = (modal) => modal.contains(event.target);
+            if (Array.from(modals).some(wasClicked)) {
+                callback();
+            }
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener("click", handleClick);
+
+        return () => {
+            document.removeEventListener("click", handleClick);
+        };
+    });
+};
+
+function base64ToHex(str64) {
```

Comment:
Nitpick: I disagree slightly with this abstraction. I think it would be better to just hex encoding function.

It won't even make the code longer:

`toHex(atob(str))` vs
`base64ToHex(str)`